Content.Server/Cloning/CloningConsoleSystem.cs

```diff
@@ -0,0 +1,261 @@
+        {
+            if (!Resolve(console, ref consoleComp))
+                return;
+
+            if (scanner != null)
+            {
+                Transform((EntityUid) scanner).Coordinates.TryDistance(EntityManager, Transform((console)).Coordinates, out float scannerDistance);
+                consoleComp.GeneticScannerInRange = scannerDistance <= consoleComp.MaxDistance;
+            }
+            if (cloningPod != null)
+            {
+                Transform((EntityUid) cloningPod).Coordinates.TryDistance(EntityManager, Transform((console)).Coordinates, out float podDistance);
+                consoleComp.CloningPodInRange = podDistance <= consoleComp.MaxDistance;
+            }
+
+            UpdateUserInterface(consoleComp);
+        }
+        private CloningConsoleBoundUserInterfaceState GetUserInterfaceState(CloningConsoleComponent consoleComponent)
+        {
+            ClonerStatus clonerStatus = ClonerStatus.Ready;
+
+            // genetic scanner info
+            string scanBodyInfo = "Unknown";
+            bool scannerConnected = false;
+            bool scannerInRange = consoleComponent.GeneticScannerInRange;
+            if (consoleComponent.GeneticScanner != null && TryComp<MedicalScannerComponent>(consoleComponent.GeneticScanner, out var scanner)) {
+
+                scannerConnected = true;
+                EntityUid? scanBody = scanner.BodyContainer.ContainedEntity;
+
+                // GET STATE
+                if (scanBody == null || !HasComp<MobStateComponent>(scanBody))
+                    clonerStatus = ClonerStatus.ScannerEmpty;
+                else
+                {
+                    scanBodyInfo = MetaData((EntityUid) scanBody).EntityName;
+
+                    TryComp<MindComponent>(scanBody, out var mindComp);
+
+                    if (!_mobStateSystem.IsDead((EntityUid) scanBody))
+                    {
+                        clonerStatus = ClonerStatus.ScannerOccupantAlive;
+                    }
+                    else
+                    {
+                        if (mindComp == null || mindComp.Mind == null || mindComp.Mind.UserId == null || !_playerManager.TryGetSessionById(mindComp.Mind.UserId.Value, out var client))
+                        {
+                            clonerStatus = ClonerStatus.NoMindDetected;
+                        }
+                    }
+                }
+            }
+
+            // cloning pod info
+            var cloneBodyInfo = "Unknown";
+            float cloningProgress = 0;
+            float cloningTime = 30f;
+            bool clonerProgressing = false;
+            bool clonerConnected = false;
+            bool clonerMindPresent = false;
+            bool clonerInRange = consoleComponent.CloningPodInRange;
+            if (consoleComponent.CloningPod != null && TryComp<CloningPodComponent>(consoleComponent.CloningPod, out var clonePod)
+            && Transform((EntityUid) consoleComponent.CloningPod).Anchored)
```

Comment:
```suggestion
            && Transform(consoleComponent.CloningPod.Value).Anchored)
```
etc for all other cases of that